test-data/typecheck/test_decorators.yml

```diff
@@ -3,22 +3,17 @@
         from rest_framework.decorators import api_view
         @api_view()
         def view_func1(request): ...
-        reveal_type(view_func1)  # N: Revealed type is 'def (request: Any) -> Any'
+        reveal_type(view_func1)  # N: Revealed type is 'rest_framework.views.AsView'
 -   case: api_view_fancy
     main: |
         from rest_framework.decorators import api_view
         from rest_framework.request import Request
         from rest_framework.response import Response
         @api_view(['GET', 'POST'])
         def view_func2(request: Request, arg: str) -> Response: ...
-        reveal_type(view_func2)  # N: Revealed type is 'def (request: rest_framework.request.Request, arg: builtins.str) -> rest_framework.response.Response'
+        reveal_type(view_func2)  # N: Revealed type is 'rest_framework.views.AsView'
 -   case: api_view_bare_is_error
     main: |
         from rest_framework.decorators import api_view
         @api_view  # E: Argument 1 to "api_view" has incompatible type "Callable[[Any], Any]"; expected "Optional[Sequence[str]]"
         def view_func2(request): ...
--   case: api_view_incorrect_return
-    main: |
-        from rest_framework.decorators import api_view
-        @api_view()  # E: Value of type variable "_VIEW" of function cannot be "Callable[[Any], List[Any]]"
-        def view_func2(request) -> list: ...
```

Comment:
I think this was a useful feature and test case: the ability to detect `@api_view` decorator on wrongly typed functions. Simply removed with no discussion.